src/org/zaproxy/zap/extension/autoupdate/AutoUpdateAPI.java

```diff
@@ -74,12 +119,55 @@ public ApiResponse handleApiView(String name, JSONObject params)
 			result = new ApiResponseElement(name, this.getLatestVersionNumber());
 		} else if (VIEW_IS_LATEST_VERSION.equals(name)) {
 			result = new ApiResponseElement(name, Boolean.toString(this.isLatestVersion()));
+		} else if (VIEW_INSTALLED_ADDONS.equals(name)) {
+	        final ApiResponseList resultList = new ApiResponseList(name);
+	        for (AddOn ao : extension.getInstalledAddOns()) {
+	            resultList.addItem(addonToSet(ao));
+	        }
+	        result = resultList;
+        } else if (VIEW_NEW_ADDONS.equals(name)) {
+            final ApiResponseList resultList = new ApiResponseList(name);
+            for (AddOn ao : extension.getNewAddOns()) {
+                resultList.addItem(addonToSet(ao));
+            }
+            result = resultList;
+        } else if (VIEW_UPDATED_ADDONS.equals(name)) {
+            final ApiResponseList resultList = new ApiResponseList(name);
+            for (AddOn ao : extension.getUpdatedAddOns()) {
+                resultList.addItem(addonToSet(ao));
+            }
+            result = resultList;
+        } else if (VIEW_MARKETPLACE_ADDONS.equals(name)) {
+            final ApiResponseList resultList = new ApiResponseList(name);
+            for (AddOn ao : extension.getMarketplaceAddOns()) {
+                resultList.addItem(addonToSet(ao));
+            }
+            result = resultList;
 		} else {
 			throw new ApiException(ApiException.Type.BAD_VIEW);
 		}
 		return result;
 	}
 	
+	private ApiResponseSet addonToSet(AddOn ao) {
+        Map<String, String> map = new HashMap<>();
+        map.put("id", String.valueOf(ao.getId()));
```

Comment:
Except for the size field all `String.valueOf()` calls should be removed, they are not really needed.